diff --git a/dist/shared/mkdist.08e478e6.mjs b/dist/shared/mkdist.08e478e6.mjs
index d811db2ef99378787d21fda666779abe53e7db2e..de77c0851783cfd9982c5a1a4c99371a9c1cb3ee 100644
--- a/dist/shared/mkdist.08e478e6.mjs
+++ b/dist/shared/mkdist.08e478e6.mjs
@@ -254,7 +254,7 @@ async function mkdist(options = {}) {
   const esmResolveExtensions = ["", "/index.mjs", "/index.js", ".mjs", ".ts"];
   for (const output of outputs.filter((o) => o.extension === ".mjs")) {
     output.contents = output.contents.replace(
-      /(import|export)(.* from ["'])(.*)(["'])/g,
+      /(import|export)([^'"]*["'])(.*)(["'])/g,
       (_, type, head, id, tail) => type + head + resolveId(output.path, id, esmResolveExtensions) + tail
     );
   }
diff --git a/dist/shared/mkdist.beb433dd.cjs b/dist/shared/mkdist.beb433dd.cjs
index 8b75d12e3091da180468c215c7ba0661e8cbbd1c..aa2222943350117bec788f9e5658b641e8e82f44 100644
--- a/dist/shared/mkdist.beb433dd.cjs
+++ b/dist/shared/mkdist.beb433dd.cjs
@@ -256,7 +256,7 @@ async function mkdist(options = {}) {
   const esmResolveExtensions = ["", "/index.mjs", "/index.js", ".mjs", ".ts"];
   for (const output of outputs.filter((o) => o.extension === ".mjs")) {
     output.contents = output.contents.replace(
-      /(import|export)(.* from ["'])(.*)(["'])/g,
+      /(import|export)([^'"]*["'])(.*)(["'])/g,
       (_, type, head, id, tail) => type + head + resolveId(output.path, id, esmResolveExtensions) + tail
     );
   }
